includes/validation/Taxonomy.php

```diff
@@ -0,0 +1,175 @@
+<?php
+namespace Press_Sync\validation;
+
+/**
+ * Class Taxonomy
+ *
+ * @package Press_Sync\validation
+ * @since   NEXT
+ */
+class Taxonomy implements CountInterface {
+	/**
+	 * Get the number of unique taxonomies in this WordPress installation.
+	 *
+	 * @return int
+	 * @since NEXT
+	 */
+	public function get_unique_taxonomy_count() {
+		return count( get_taxonomies() );
+	}
+
+	/**
+	 * Get the number of terms for each taxonomy in this WordPress installation.
+	 *
+	 * @return array
+	 * @since NEXT
+	 */
+	public function get_term_count_by_taxonomy() {
+		$terms = [];
+
+		foreach ( get_taxonomies() as $name => $taxonomy ) {
+			$terms[] = array(
+				'taxonomy_name'   => $name,
+				'number_of_terms' => count(
+					get_terms(
+						array(
+							'taxonomy'   => $taxonomy,
+							'hide_empty' => false,
+						)
+					) ),
+			);
+		}
+
+		return $terms;
+	}
+
+	/**
+	 * Get an indexed array of post counts by taxonomy and term.
+	 *
+	 * @TODO Let's split up some of these responsibilities.
+	 *
+	 * @return array
+	 * @since NEXT
+	 */
+	public function get_post_count_by_taxonomy_term() {
+		$taxonomies = array();
+
+		foreach ( get_taxonomies() as $name => $taxonomy ) {
+			$taxonomies[ $name ] = array();
+
+			foreach (
+				get_terms(
```

Comment:
I would prefer to see the output of `get_terms` stored in a variable instead of having the function call in the `foreach` itself.

Reply:
I'm personally not a fan of assigning variables where they'll only get used once, but this is easy enough to adjust and I agree it makes it more readable. Fixed.